source/validate_cfg.cpp

```diff
@@ -268,21 +271,52 @@ void UpdateContinueConstructExitBlocks(
   }
 }
 
+tuple<string, string, string> ConstructNames(ConstructType type) {
+  string construct_name, header_name, exit_name;
+
+  switch (type) {
+    case ConstructType::kSelection:
+      construct_name = "selection";
+      header_name = "selection header";
+      exit_name = "merge block";
+      break;
+    case ConstructType::kLoop:
+      construct_name = "loop";
+      header_name = "loop header";
+      exit_name = "merge block";
+      break;
+    case ConstructType::kContinue:
+      construct_name = "continue";
+      header_name = "continue target";
+      exit_name = "back-edge block";
+      break;
+    case ConstructType::kCase:
+      construct_name = "case";
+      header_name = "case block";
```

Comment:
Optional:  How about "case entry block" and "case exit block". ?    If I feel strongly about this I can change it in a subsequent CL.